src/bp/ui-admin/src/Pages/Bot/Details.jsx

```diff
@@ -41,7 +42,9 @@ class Bots extends Component {
     categories: [],
     moreOpen: false,
     languages: [],
-    defaultLanguage: undefined
+    defaultLanguage: undefined,
+    formHasChanged: false,
```

Comment:
Just so you know this could be avoided by computing changes from the state and the initial bot values, making this property obsolete. I assume you thought of it and took the decision that it's simpler to store a modifiedState.

Reply:
tbh I just didn't think of it 🤦‍♂ I must've been tired 😂 I'll do that and it'll address your next comment about changes being reverted back to their original state